### PR TITLE
SSRS two machine initial commit

### PR DESCRIPTION
**Pull request by jtarquino**
--------------------MESSAGE FROM ADMIN, DELETE BEFORE SUBMITTING----------------------

Thank you for deciding to contribute to this repository, we're thrilled to have you contribute a template or a fix! :tada: Before we proceed, please be sure to review the contribution guidelines that will make the review process go smoother.
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

We've created an outline of recommended sections to fill out that will help make this Pull Request awesome!

--------------------MESSAGE FROM ADMIN, DELETE BEFORE SUBMITTING----------------------

### Changelog
* Template 
* Parameters
* Metadata
* Readme

### Description of the change
Adding the SSRS two machine non domain join template

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@jtarquino__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
Hi __@jtarquino__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: sql-reportingservices-two-machines/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+        "description": "Username for SA of the database"
+      }
+    },
+    "dbSAUserPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for SA of the database"
+      }
+    },
+    "dnsNameForPublicIP": {
```
Comment (singhkay) [inline sql-reportingservices-two-machines/azuredeploy.json]: Call this dnsPrefix instead. See #966 for guidance

File: sql-reportingservices-two-machines/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+    "wadmetricsresourceid": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name , '/providers/', 'Microsoft.Compute/virtualMachines/')]",
+    "wadcfgxend": "\"><MetricAggregation scheduledTransferPeriod=\"PT1H\"/><MetricAggregation scheduledTransferPeriod=\"PT1M\"/></Metrics></DiagnosticMonitorConfiguration></WadCfg>"
+
+
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('vhdStorageNameRs')]",
+      "apiVersion": "2015-06-15",
```
Comment (singhkay) [inline sql-reportingservices-two-machines/azuredeploy.json]: Make api version a variable that is reused across the resources

File: sql-reportingservices-two-machines/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+    "vmImageOfferRs": "[variables('sqlImageOffer')]",
+    "vmImageSkuRs": "[variables('sqlImageSku')]",
+    "vmSizeDataSource": "[parameters('vmSize')]",
+    "vmNameDataSource": "[concat(parameters('vmPrefix'), '-DataSource')]",
+    "vmSizeCatalog": "[parameters('vmSize')]",
+    "vmNameCatalog": "[concat(parameters('vmPrefix'), '-Catalog')]",
+    "virtualNetworkName": "SSRSVNET",
+    "vnetId": "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetId'), '/subnets/', variables('subnetName'))]",
+    "vhdStorageNameRs": "[concat('vhdrs', uniqueString(resourceGroup().id))]",
```
Comment (singhkay) [inline sql-reportingservices-two-machines/azuredeploy.json]: Do the uniquestring first in concats. It ensures the storage accounts generated from this template will be on different storage stamps

File: sql-reportingservices-two-machines/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,553 @@
...
+    "vhdStorageNameDataSource": "[concat('vhdds', uniqueString(resourceGroup().id))]",
+    "vhdStorageNameCatalog": "[concat('vhdcatalog', uniqueString(resourceGroup().id))]",
+    "catalogPublicIpId": "[concat('Microsoft.Network/publicIPAddresses/',concat(variables('publicIPAddressName'), 0))]",
+    "dnsName": "[parameters('dnsNameForPublicIP')]",
+    "sqlConfigurationConfigurationFunction": "PrepareSqlServer.ps1\\PrepareSqlServer",
+    "sqlSAConfigurationConfigurationFunction": "PrepareSqlServerSa.ps1\\PrepareSqlServerSa",
+    "rsConfigurationConfigurationFunction": "PrepareRsServer.ps1\\PrepareSSRSServer",
+    "sqlModuleUrl": "https://raw.githubusercontent.com/jtarquino/RsArm/master/RSARM/DSC/PrepareSqlServer.ps1.zip",
```
Comment (singhkay) [inline sql-reportingservices-two-machines/azuredeploy.json]: This should point to the module in this repo. You can put Azure link in here and the CI will replace this to your repo's location for test. See https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#travis-ci

File: sql-reportingservices-two-machines/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,553 @@
...
+      "apiVersion": "2015-06-15",
+      "tags": {
+          "displayName": "DiagnosticsStorageAccount"
+      },
+      "properties": {
+          "accountType": "Standard_LRS"
+      }
+      },
+    {
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline sql-reportingservices-two-machines/azuredeploy.json]: Can use the GA API 2015-06-15 for all compute, network and storage resources

File: sql-reportingservices-two-machines/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,553 @@
...
+            "[concat('Microsoft.Compute/virtualMachines/', variables('vmNameRs'))]",
+            "[resourceId('Microsoft.Compute/virtualMachines/extensions', variables('vmNameCatalog'), 'SQLConfigurationMixedAuth')]"
+          ],
+          "tags": {
+            "displayName": "SSRSConfiguration"
+          },
+          "properties": {
+            "publisher": "Microsoft.Powershell",
+            "type": "DSC",
+            "typeHandlerVersion": "2.9",
```
Comment (singhkay) [inline sql-reportingservices-two-machines/azuredeploy.json]: DSC is up to 2.14 now

File: sql-reportingservices-two-machines/azuredeploy.json (hunk 2 above)
Comment (jtarquino) [inline sql-reportingservices-two-machines/azuredeploy.json]: Will do, you might want to reach visual studio as it complains about it
Value must be one of the following values: "2015-05-01-preview", "2015-06-15"

**Comment by singhkay** [comment]
@jtarquino CI passed, looking good! 

One minor fix before merging. I just noticed the name of the folder is `sql-reportingservices-two-machines`. One suggestion here is it should be `sql-reporting-services-*` for better readability. Also is `two-machines` really needed? I think it's OK to leave it out

**Comment by jtarquino** [comment]
Regarding the last comment about the two machines, that is the key difference of the template that splits reporting services and the database in different machines and leave it ready to use, do you have a suggestion for the folder based on that?

**Comment by singhkay** [comment]
@jtarquino Not atm. `two-machines` just makes it sound like maybe this is setup in a HA way with replicated VMs. Also makes it sounds like there maybe a 3 vm template coming next. But this disambiguation about splitting the db and reporting services into two vms could also be better left for the readme and go with `sql-reporting-services-sql-server` as the name for the folder

**Comment by singhkay** [comment]
@jtarquino Forgot to add, don't forget to change the links in the readme :)

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**